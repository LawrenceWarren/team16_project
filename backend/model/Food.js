const mongoose = require("mongoose");

const Schema = mongoose.Schema;

const foodSchema = new Schema({
  _id: mongoose.Schema.Types.ObjectId,
  image: { type: String, required: true },
  name: { type: String, required: true },
  address: { type: String, required: true },
  type: { type: String, required: true },
  price: { type: String, required: true },
  link: { type: String, required: true }
});

const Food = mongoose.model("food", foodSchema, "foods");

module.exports = Food;
