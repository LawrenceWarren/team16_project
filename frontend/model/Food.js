const mongoose = require("mongoose");
const Schema = mongoose.Schema;

//Create schema
const foodSchema = new Schema({
  image: {
    type: String,
    required: true,
  },
  name: {
    type: String,
    required: true,
  },
  address: {
    type: String,
    required: true,
  },
  type: {
    type: String,
    required: true,
  },
  price: {
    type: String,
    required: true,
  },
  link: {
    type: String,
    required: true,
  },
  registerDate: {
    type: Date,
    required: true,
    default: new Date(),
  },
});

//Create Model
//"Food" is correspond to the collection name in database for foods
const Food = mongoose.model("food", foodSchema);
module.exports = Food;
