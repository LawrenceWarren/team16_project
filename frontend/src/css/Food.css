.picture {
  max-width: 40%;
  width: 40%;
  max-height: 50%;
  height: 50%;
  margin: "100vw";
  display: inline-block;
}

.imgButton {
  display: inline-block;
}

.buttonRow {
  text-align: center;
}

.content {
  text-align: center;
}

.title {
  text-align: center;
}