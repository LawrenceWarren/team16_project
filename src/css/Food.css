/*mainDiv is the parent div of all others in food*/
.mainDiv {
  text-align: center;
  max-width: 100vw;
}

/*The pink banner across the top of the food page*/
.banner {
  background-color: #c0087f;
  margin-top: 1vw;
  margin-bottom: 1.5vw;
  color: white;
  font-size: 2.5vw;
  font-family: "Microsoft YaHei", 微软雅黑, "MicrosoftJhengHei", 华文细黑,
    STHeiti, MingLiu;
  text-align: center;
}

/*Styling for the container of each item*/
.contentContainer {
  background-color: white;
  border-radius: 20px;
  border: 0.25vw solid #c0087f;
  margin: 1.5vw;
  text-align: center;
  display: inline-block;
  min-width: 350px;
  width: 85%;
  height: 70vh;
  max-width: 85%;
  padding-left: 0px;
  padding-right: 0px;
  box-shadow: 5px #000000;
}

/*Styling for left and right on screen swipe indicators*/
.swipeRight {
  height: auto;
  width: 10%;
  float: right;
  padding: 10px;
}
.swipeLeft {
  height: auto;
  width: 10%;
  float: left;
  padding: 10px;
}

/*Styling for the picture for each item*/
.picture {
  max-width: 70%;
  height: auto;
  margin: 10px;
  display: inline-block;
  border-radius: 5px;
}

/*Basic styling for content & titling of each items information*/
.content,
.title {
  text-align: center;
  font-family: "Microsoft YaHei", 微软雅黑, "MicrosoftJhengHei", 华文细黑,
    STHeiti, MingLiu;
  line-height: 105%;
}
