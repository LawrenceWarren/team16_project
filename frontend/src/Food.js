import React from "react";
import "./css/Food.css";
import Header from "./Header";
import Footer from "./Footer";

class Food extends React.Component {
  constructor() {
    super();

    this.leftSwipe = this.onClickBack.bind(this);
    this.rightSwipe = this.onClickForward.bind(this);

    this.state = {
      index: 0,
      foodList: []
    };
  }

  //Handle button clicks forward and back
  //TODO: make work for swipes on screen
  onClickForward = () => {
    if (this.state.index + 1 === this.state.foodList.length) {
      this.setState({
        index: 0
      });
    } else {
      this.setState({
        index: this.state.index + 1
      });
    }
  };
  onClickBack = () => {
    if (this.state.index - 1 === -1) {
      this.setState({
        index: this.state.foodList.length - 1
      });
    } else {
      this.setState({
        index: this.state.index - 1
      });
    }
  };

  //Calls the server upon page loading
  callServer() {
    fetch("http://localhost:4000/food/")
      .then(res => res.json())
      .then(res => this.setState({ foodList: res }))
      .catch(err => err);
  }

  //Mounts the callServer function upon page loading, calling it
  componentWillMount() {
    this.callServer();
  }

  //!-------------!
  //!Render method!
  //!-------------!
  render() {
    console.log(
      "Debug:\n" + JSON.stringify(this.state.foodList[this.state.index]?.name)
    );

    return (
      <div class="mainDiv">
        <Header />
        <br />
        <br />

        <button class="imgButton" onClick={this.onClickBack}>
          {"<="}
        </button>

        <div class="contentContainer">
          <img
            src={this.state.foodList[this.state.index]?.image}
            class="picture"
            alt=""
          />

          <b>
            <p class="title">Name</p>
          </b>
          <p class="content">{this.state.foodList[this.state.index]?.name}</p>

          <b>
            <p class="title">Address</p>
          </b>
          <p class="content">
            {this.state.foodList[this.state.index]?.address}
          </p>

          <b>
            <p class="title">Type</p>
          </b>
          <p class="content">{this.state.foodList[this.state.index]?.type}</p>
          <b>
            <p class="title">Price</p>
          </b>
          <p class="content">{this.state.foodList[this.state.index]?.price}</p>

          <b>
            <p class="title">External Link</p>
          </b>
          <a href={this.state.foodList[this.state.index]?.link}>
            <p class="content">{this.state.foodList[this.state.index]?.link}</p>
          </a>
        </div>
        <button class="imgButton" onClick={this.onClickForward}>
          {"=>"}
        </button>

        <Footer />
      </div>
    );
  }
}

export default Food;
