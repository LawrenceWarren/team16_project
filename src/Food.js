import React from "react"; //We're writing React code. Duh.
import "./css/Food.css"; //Style sheet
import Header from "./Header"; //The header of the page
import Footer from "./Footer"; //The footer of the page
import { Swipeable } from "react-swipeable"; //Used to create a swipeable DOM element

class Food extends React.Component {
  constructor() {
    super(); //Does something ??

    this.leftSwipe = this.onClickBack.bind(this); //Binds the onClickBack function on page loading
    this.rightSwipe = this.onClickForward.bind(this); //Binds the onClickFroward function on page load

    /*
    State is populated with changing features of the page:
     * an array of eateries
     * an index for that array
    */
    this.state = {
      index: 0, //Used for indexing the array
      foodList: [
        //Food list
        {
          image: require("./resource/food/connectionFail.png"), //Default value
          name: "",
          address: "",
          type: "",
          price: "",
          link: "",
        },
      ],
    };
  }

  //Handle swipes forward
  //Sets the state of index to increase by 1,
  // or wrap around to 0
  onClickForward = () => {
    if (this.state.index + 1 === this.state.foodList.length) {
      this.setState({
        index: 0,
      });
    } else {
      this.setState({
        index: this.state.index + 1,
      });
    }
  };
  //Handles swipe back
  //Sets the state of index to decrease by 1,
  //or wrap around to the largest value
  onClickBack = () => {
    if (this.state.index === 0) {
      this.setState({
        index: this.state.foodList.length - 1,
      });
    } else {
      this.setState({
        index: this.state.index - 1,
      });
    }
  };

  //Calls the server after the DOM is rendered
  async componentDidMount() {
    console.log("Food: Fetching from server...");
    try {
      const res = await fetch("/foodReq");
      if (res.status >= 400) {
        throw new Error("There was an error in the HTTP request.");
      }

      const food = await res.json();
      this.setState({ foodList: food });
      console.log("Food: Data from the server has been received!");
    } catch (err) {
      console.error("Food: " + err);
    }
  }

  //!-------------!//
  //!Render method!//
  //!-------------!//
  render() {
    //!DOM
    return (
      <div className="mainDiv">
        <Header /> {/*The header component written by Steven*/}
        <p className="banner"> Eateries around the hospital </p>
        {/*Creates a swipeable div element, and binds the swipe handler functions to swipe events
         * However, swipeable does not allow for styling, hence the immediately nested div*/}
        <Swipeable
          onSwipedLeft={this.onClickBack}
          onSwipedRight={this.onClickForward}
        >
          {/*Styled content container, within which images and text appear*/}
          <div className="contentContainer">
            {/*On screen indications for swiping */}
            <img
              src={require("./resource/food/swipeLeft.png")}
              className="swipeLeft"
              onClick={this.onClickBack}
              alt="Upon pressing, you move backwards through the eateries."
            />
            <img
              src={require("./resource/food/swipeRight.png")}
              className="swipeRight"
              onClick={this.onClickForward}
              alt="Upon pressing, you move forwards through the eateries."
            />
            {/*Picture */}
            <img
              src={this.state.foodList[this.state.index]?.image}
              className="picture"
              alt=""
            />
            {/*Name */}
            <b>
              <u>
                <p className="title">Name</p>
              </u>
            </b>
            <p id="nameID" className="content">
              {this.state.foodList[this.state.index]?.name}
            </p>
            {/*Address*/}
            <b>
              <u>
                <p className="title">Address</p>
              </u>
            </b>
            <p id="addressID" className="content">
              {this.state.foodList[this.state.index]?.address}
            </p>
            {/*Type*/}
            <b>
              <u>
                <p className="title">Type</p>
              </u>
            </b>
            <p id="typeID" className="content">
              {this.state.foodList[this.state.index]?.type}
            </p>
            {/*Pricing*/}
            <b>
              <u>
                <p className="title">Price</p>
              </u>
            </b>
            <p id="priceID" className="content">
              {this.state.foodList[this.state.index]?.price}
            </p>
            {/*Link to site*/}
            <b>
              <u>
                <p className="title">External Link</p>
              </u>
            </b>
            <a href={this.state.foodList[this.state.index]?.link}>
              <p id="linkID" className="content">
                {this.state.foodList[this.state.index]?.link}
              </p>
            </a>
          </div>
        </Swipeable>
        <Footer /> {/*Footer component written by Steven*/}
      </div>
    );
  }
}

export default Food;
