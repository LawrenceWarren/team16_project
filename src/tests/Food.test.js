import React from "react";
import { create } from "react-test-renderer";
import Food from "../Food";

describe("Food Page renders:", () => {
  test("Without failing", () => {
    const button = create(<Food />);
    expect(button.toJSON()).toMatchSnapshot();
  });
});
