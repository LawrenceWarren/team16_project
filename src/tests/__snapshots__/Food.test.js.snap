// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food Page renders: Without failing 1`] = `
<div
  className="mainDiv"
>
  <div>
    <div
      className=""
      id="fixed-menu"
    >
      <div
        className="headerUl"
      >
        <li
          className=""
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <a
            className="menulevel"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <div
            className="submenu-hidden"
          />
        </li>
        <li
          className=""
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <span
            className="menulevel"
          >
            Infrastructure
          </span>
          <div
            className="submenu-hidden"
          >
            <a
              className=""
              data-id={0}
              href="/Food/"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
            >
              Food
            </a>
            <a
              className=""
              data-id={1}
              href="/Charity/"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
            >
              Charities
            </a>
            <a
              className=""
              data-id={2}
              href="/Accommodation/"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
            >
              Accommodation
            </a>
          </div>
        </li>
        <li
          className=""
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <span
            className="menulevel"
          >
            Contact
          </span>
          <div
            className="submenu-hidden"
          >
            <a
              className=""
              data-id={0}
              href="https://www.facebook.com/avas.angels.739"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Facebook
            </a>
            <a
              className=""
              data-id={1}
              href="https://twitter.com/AvasAngels_com"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Twitter
            </a>
            <a
              className=""
              data-id={2}
              href="https://www.instagram.com/avasangelscharity/"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Instagram
            </a>
            <a
              className=""
              data-id={3}
              href="https://www.linkedin.com/in/avas-angels-1519a2160/"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              LinkedIn
            </a>
            <a
              className=""
              data-id={4}
              href="/Contact/"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
            >
              Leave Contact Info
            </a>
          </div>
        </li>
        <li
          className=""
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <span
            className="menulevel"
          >
            Get Involved
          </span>
          <div
            className="submenu-hidden"
          >
            <a
              className=""
              data-id={0}
              href="https://www.paypal.com/fundraiser/112574636177901026/charity/3575409"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Donate
            </a>
            <a
              className=""
              data-id={1}
              href="https://www.avas-angels.com/getinvolved.html"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Volunteers
            </a>
            <a
              className=""
              data-id={2}
              href="https://www.avas-angels.com/getinvolved.html"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Supporters
            </a>
            <a
              className=""
              data-id={3}
              href="https://www.avas-angels.com/getinvolved.html"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Sponsors
            </a>
            <a
              className=""
              data-id={4}
              href="/Feedback/"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
            >
              FeedBack
            </a>
          </div>
        </li>
        <li
          className=""
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
        >
          <span
            className="menulevel"
          >
            More on Ava's Angels
          </span>
          <div
            className="submenu-hidden"
          >
            <a
              className=""
              data-id={0}
              href="/About/"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
            >
              About Ava's Angels
            </a>
            <a
              className=""
              data-id={1}
              href="https://www.avas-angels.com/angelservices.html"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Angel Services
            </a>
            <a
              className=""
              data-id={2}
              href="https://www.avas-angels.com/difference.html"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              How we make a difference
            </a>
            <a
              className=""
              data-id={3}
              href="https://www.avas-angels.com/supporters.html"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Proud Supporters
            </a>
            <a
              className=""
              data-id={4}
              href="https://www.avas-angels.com/shop.html"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              Shop
            </a>
            <a
              className=""
              data-id={5}
              href="https://www.avas-angels.com/news.html"
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              rel="noopener noreferrer"
              target="_blank"
            >
              News/Press
            </a>
          </div>
        </li>
      </div>
    </div>
    <header>
      <div
        align="center"
        className="pageHeader"
      >
        <img
          alt="Ava's Angel"
          src="https://www.avas-angels.com/images/HiResLogo.png"
        />
      </div>
    </header>
  </div>
   
  <p
    className="banner"
  >
     Eateries around the hospital 
  </p>
  <div>
    <div
      className="contentContainer"
    >
      <img
        className="swipeLeft"
        onClick={[Function]}
        src="swipeLeft.png"
      />
      <img
        className="swipeRight"
        onClick={[Function]}
        src="swipeRight.png"
      />
      <img
        alt=""
        className="picture"
        src="connectionFail.png"
      />
      <b>
        <p
          className="title"
        >
          Name
        </p>
      </b>
      <p
        className="content"
      >
        
      </p>
      <b>
        <p
          className="title"
        >
          Address
        </p>
      </b>
      <p
        className="content"
      >
        
      </p>
      <b>
        <p
          className="title"
        >
          Type
        </p>
      </b>
      <p
        className="content"
      >
        
      </p>
      <b>
        <p
          className="title"
        >
          Price
        </p>
      </b>
      <p
        className="content"
      >
        
      </p>
      <b>
        <p
          className="title"
        >
          External Link
        </p>
      </b>
      <a
        href=""
      >
        <p
          className="content"
        >
          
        </p>
      </a>
    </div>
  </div>
  <div
    className="footerMain"
  >
    <p
      className="banner"
    >
       Get Involved: 
    </p>
    <div
      align="center"
    >
      <button
        className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-1 MuiButton-contained MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Become Part Of Ava's Angels
        </span>
      </button>
    </div>
    <button
      className="donate"
      onClick={[Function]}
    />
    <div
      className="SocialButtons"
    >
      <button
        className="butt fb-butt"
        onClick={[Function]}
      />
      <button
        className="butt tw-butt"
        onClick={[Function]}
      />
      <button
        className="butt ins-butt"
        onClick={[Function]}
      />
      <button
        className="butt li-butt"
        onClick={[Function]}
      />
      <button
        className="butt feed-butt"
        onClick={[Function]}
      />
    </div>
    <div
      className="furtherInfo"
    >
      | 
      <a
        href="https://www.google.com"
      >
         FAQ 
      </a>
       | 
      <a
        href="https://www.google.com"
      >
         Terms of Use 
      </a>
       |
      <a
        href="https://www.google.com"
      >
         Privacy 
      </a>
       |
      <a
        href="https://www.google.com"
      >
         Contact 
      </a>
       |
    </div>
    <div
      className="copyRight"
    >
      <p>
        © Copyright 2019-2020. Designed and powered by Ali S., Lawrence W., Shan W.Z., Benjamin S., Chen Y.T. and Zhang J.H. All Rights Reserved.
         
      </p>
    </div>
  </div>
   
</div>
`;
